File: components/appexample/src/main/java/com/ecfront/dew/appexample/controller/TestController.java
package com.ecfront.dew.appexample.controller;

import com.ecfront.dew.appexample.service.HystrixService;
import com.ecfront.dew.common.Resp;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping(value = "/test")
public class TestController {

    private static final Logger logger = LoggerFactory.getLogger(TestController.class);

    @Autowired
    private HystrixService hystrixService;

    @GetMapping(value = "")
    @ResponseBody
    public Resp<String> get() throws Exception {
        return hystrixService.doGet();
    }

    @GetMapping(value = "error")
    @ResponseBody
    public Resp<String> error() throws Exception {
        return hystrixService.errorGet();
    }

}

File: framework/components/auth/src/test/scala/com/tairanchina/csp/dew/auth/TestController.scala
package com.tairanchina.csp.dew.auth

import com.ecfront.dew.common.Resp
import com.tairanchina.csp.dew.auth.service.BasicService
import io.swagger.annotations.Api
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation._

@RestController
@Api(value = "测试")
@RequestMapping(value = Array("/test"))
@Validated
class TestController @Autowired()(
                                   val basicService: BasicService
                                 ) {

  @GetMapping(Array(""))
  def test1(): Resp[Void] = {
    Resp.success(null)
  }

  @GetMapping(Array("/auth"))
  def test2(): Resp[Void] = {
    Resp.success(null)
  }

  @GetMapping(Array("/auth/t"))
  def test3(): Resp[Void] = {
    Resp.success(null)
  }

}
